src/org/labkey/test/tests/FileAttachmentColumnTest.java

```diff
@@ -15,153 +15,609 @@
+
+        for (File file : SAMPLE_FILES)
+        {
+            var queryUpdatePage = samplesRegion.clickInsertNewRow();
+            queryUpdatePage.setField("Name", file.getName());
+            queryUpdatePage.setField("file", file)
+                    .submit();
+        }
+        validateSampleData(SUBA_SAMPLETYPE_NAME, SUBFOLDER_A_PATH, SAMPLE_FILES);
+
+        var fileContentPage = FileContentPage.beginAt(this, SUBFOLDER_A_PATH);
+        fileContentPage.fileBrowserHelper().selectFileBrowserItem("/sampletype/csv_sample.csv");
+        var files = fileContentPage.fileBrowserHelper().getFileList(true);
+        checker().withScreenshot("unexpected_davfile_state")
+                .wrapAssertion(()-> Assertions.assertThat(files)
+                        .as("expect edited files to be present in dav locaiton")
+                        .containsExactlyInAnyOrderElementsOf(fileNames));
     }
 
-    @BeforeClass
-    public static void setupProject()
+    // assay
+    @Test
+    public void testAssayFileFieldsUI() throws Exception
     {
-        FileAttachmentColumnTest init = (FileAttachmentColumnTest)getCurrentTest();
-        init.doSetup();
+        String runName = "assay_run.tsv";
+
+        // generate some resultFiles
+        var resultFiles = new ArrayList<File>();
+        WebDavUploadHelper uploadHelper = new WebDavUploadHelper(SUBFOLDER_A_PATH);
+        List<Map<String, Object>> importData = new ArrayList<>();
+        for (int i = 0; i < 5; i++)
+        {
+            String fileName = String.format("field_file_for_results_domain-%d.tsv", i);
+            String result = String.format("result-%d", i);
+            String fileText = "resultTxt\tresultFile\totherResultFile\n"+
+                        result+"\t" + fileName +"\t" + SAMPLE_FILES.get(i).getName();
+            var fieldFile = TestFileUtils.writeTempFile(fileName, fileText);
+            uploadHelper.uploadFile(fieldFile);
+            uploadHelper.uploadFile(SAMPLE_FILES.get(i));
+            resultFiles.add(fieldFile);
+
+            importData.add(Map.of(RESULT_TXT_COL, result, RESULT_FILE_COL, fieldFile.getName()));
+        }
+
+        // generate a run file, referencing the result files
+        String importDataFileContents = TestDataUtils.tsvStringFromRowMaps(importData,
+                List.of(RESULT_TXT_COL, RESULT_FILE_COL), true);
+        File importFile = TestFileUtils.writeTempFile(runName, importDataFileContents);
+
+        beginAt(SUBFOLDER_A_PATH + "/project-begin.view");
+        goToModule("Assay");
+        clickAndWait(Locator.linkContainingText(SUB_A_ASSAY));
+        clickButton("Import Data");
+        clickButton("Next");    // batch properties
+
+        // run properties
+        setFormElement(Locator.input("name"), runName);
+        setFormElement(Locator.input(RUN_TXT_COL), "run text");
+        setFormElement(Locator.input(RUN_FILE_COL), importFile);
```

Comment:
Could use any existing sample file for importFile. Also, could you please rename this to runFileFieldFile?